acquia_lift.admin.inc

```diff
@@ -287,36 +304,101 @@ function acquia_lift_admin_form_submit($form, &$form_state) {
-    if (empty($option_set->targeting)) {
-      return $report;
+function acquia_lift_report($form, &$form_state, $agent_data) {
+  $form = array(
+    '#prefix' => '<div id="acquia-lift-reports">',
+    '#suffix' => '</div>',
+    '#attached' => array(
+      'css' => array(
+        drupal_get_path('module', 'acquia_lift') . '/css/acquia_lift.admin.css',
+      ),
+      'js' => array(
+        drupal_get_path('module', 'acquia_lift') . '/js/acquia_lift.admin.js',
+      ),
+      'library' => array(
+        array('acquia_lift', 'acquia_lift.reports'),
+      ),
+    ),
+  );
+  if (!acquia_lift_is_targeting_agent($agent_data)) {
+    $legacy_campaigns = variable_get('acquia_lift_legacy_agents', array());
+    if (in_array($agent_data->machine_name, $legacy_campaigns)) {
+      $form['report'] = acquia_lift_report_custom($form_state, $agent_data, FALSE);
+      return $form;
     }
-    foreach($option_set->targeting as $audience_name => $audience) {
-      if (isset($audience['osid'])) {
-        if ($option_set = personalize_option_set_load($audience['osid'])) {
-          $nested_agent = personalize_agent_load($option_set->agent);
-          if (!empty($nested_agent)) {
-            $report[$nested_agent->machine_name] = array(
-              'title' => array(
-                '#markup' => '<h2>' . t('Test report for audience @audience', array('@audience' => $audience_name)) . '</h2>'
-              ),
-              'report' => acquia_lift_report($form, $form_state, $nested_agent, NULL, TRUE)
-            );
-          }
-        }
-      }
+    else {
+      return array();
     }
-    return $report;
   }
-  elseif (!$nested && acquia_lift_is_testing_agent($agent_data)) {
-    $legacy = TRUE;
+  $option_set = acquia_lift_get_option_set_for_targeting($agent_data->machine_name);
+
+  if (empty($option_set->targeting)) {
+    return array();
+  }
+  // Display a report for the specified audience.
+  if (isset($form_state['values']['audience_filter'])) {
+    $audience = $form_state['values']['audience_filter'];
+  }
+  else {
+    $audience = isset($_GET['targeting_audience']) ? $_GET['targeting_audience'] : NULL;
+  }
+  // Get audience options.
+  $audience_options = array();
+  foreach($option_set->targeting as $audience_name => $targeting) {
+    if (isset($targeting['osid'])) {
+      $audience_options[$audience_name] = $targeting['label'];
+    }
   }
 
+  if (empty($audience) && !isset($option_set->targeting[$audience]['osid'])) {
+    if (empty($audience_options)) {
+      return array(
+        'no_report' => array(
+          '#markup'=> t('This personalization does not contain tests, no reports to show.')
+        )
+      );
+    }
+    $audience = key($audience_options);
+  }
+
+  if ($option_set = personalize_option_set_load($option_set->targeting[$audience]['osid'])) {
+    $nested_agent = personalize_agent_load($option_set->agent);
+    if (!empty($nested_agent)) {
```

Comment:
If there is only one audience can we just change this to show the audience label instead of the dropdown:  Also - it would be nicer in the display if there were inline - pretty sure you can just add "container-inline" to the classes for the form item.